test/test_operations.py

```diff
@@ -1401,6 +1401,106 @@ def test_flip_empty_tensor(self):
     data = torch.tensor([])
     self.assertEqual(data, data.flip(0))
 
+  def test_roll(self):
+    device = xm.xla_device()
+    numbers = torch.arange(1, 9, device=device)
+
+    single_roll = numbers.roll(1, 0)
+    expected = torch.tensor([8, 1, 2, 3, 4, 5, 6, 7], device=device)
+    self.assertEqual(
```

Comment:
can we write test using `runAtenTest` util which will call the op on cpu and xla tensor then compare the output.

Reply:
actually I think we need a C++ test instead of the python test. Most likely pytorch already have a pytorch test for roll and we currently fall back to CPU.

Can you write a C++ test similar to https://github.com/pytorch/xla/blob/master/test/cpp/test_aten_xla_tensor.cpp#L327 ?

Reply:
done